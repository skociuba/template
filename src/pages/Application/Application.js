import React, { Suspense } from "react"
import { HashRouter as Switch } from "react-router-dom"
import routes, { RouteWithSubRoutes } from "../../routes.config"

const Application = () => {
  return (
    <div className="App">
      <Suspense fallback={<div />}>
        <Switch>
          {routes.map((route, i) => (
            <RouteWithSubRoutes key={i} {...route} />
          ))}
        </Switch>
      </Suspense>
    </div>
  )
}

Application.defaultProps = {}

export default Application
