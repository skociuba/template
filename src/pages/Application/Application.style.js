import {css} from 'emotion.instance';

export const applicationWrapper = css`
  min-height: 100vh;
  background: #6a899c;
  flex-direction: column;
  overflow: hidden;
  margin: 20px;
  position: relative;
`;
export const mobileApplicationWrapper = css`
  min-height: 100vh;
  background: #6a899c;
  flex-direction: column;
  overflow: hidden;
  margin-left: 10px;
  position: relative;
`;
