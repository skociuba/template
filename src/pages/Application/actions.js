import {createAction} from 'redux-actions';
import actionNames from 'utils/actionNames';

const CHECK_APP_CONFIG = 'CHECK_APP_CONFIG';
const SET_CONFIG = 'SET_CONFIG';

export const appActions = actionNames([CHECK_APP_CONFIG, SET_CONFIG]);

export const appCheckConfig = createAction(appActions.CHECK_APP_CONFIG);
export const appSetConfig = createAction(appActions.SET_CONFIG);
