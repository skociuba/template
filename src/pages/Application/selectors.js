import { createSelector } from "reselect"

export const applicationSelector = (state) => state.application || null

export const paramSelector = createSelector(applicationSelector, (app) => {
  const { params } = app?.config | {}
  if (!params) {
    return null
  }
  const result = {
    locale: params.localeCode,
    opUnit: params.opUnit,
    channelId: params.channelId,
    channelCountryCode: params.channelCountryCode,
    channelGroupMemberID: params.channelGroupMemberID,
    channelBusinessLine: params.channelBusinessLine,
    displayTimeZone: params.displayTimeZone,
  }
  return result
})
