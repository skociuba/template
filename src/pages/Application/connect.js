import {connect} from 'react-redux';

import {appCheckConfig} from './actions';
import Application from './Application';

export const mapStateToProps = (state) => {
  return {
    config: state.application.config,
  };
};

const mapDispatchToProps = {
  appCheckConfig,
};
export default connect(mapStateToProps, mapDispatchToProps)(Application);
