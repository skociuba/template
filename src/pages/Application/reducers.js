import {handleActions} from 'redux-actions';
//import { v4 as uuid } from "uuid"
//import { parsePayload } from "./utils/parsePayload"

import {appSetConfig} from './actions';

export const initialState = {
  config: null,
  loading: [],
};
export default handleActions(
  {
    [appSetConfig](state, {payload}) {
      return {
        ...state,
        config: {
          ...state.config,
          ...payload,
        },
      };
    },
  },
  initialState,
);
