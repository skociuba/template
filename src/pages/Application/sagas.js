import {put, all, select, takeLatest} from 'redux-saga/effects';

import {ENVS} from '../../config/constants';
import {getDynamicConfig, getOverridableConfig} from '../../config/appConfig';

import {appActions, appSetConfig} from './actions';

export function* checkConfigSaga() {
  try {
    const config = yield select(({application}) => application.config);

    if (!config) {
      const dynamicConfig = getDynamicConfig();
      const staticCalculatedConfig = {
        ...getOverridableConfig(),
      };
      const allConfigs = {...dynamicConfig, ...staticCalculatedConfig};

      yield put(appSetConfig(allConfigs));
    }
  } catch (e) {
    if (process.env.NODE_ENV !== ENVS.PRODUCTION) {
      console.error('cannot updated data');
    }
  }
}

// eslint-disable-next-line import/no-anonymous-default-export
export default function*() {
  yield all([takeLatest(appActions.CHECK_APP_CONFIG, checkConfigSaga)]);
}
