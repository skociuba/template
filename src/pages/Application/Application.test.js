import Application from './Application';

const defaultProps = {
  appCheckConfig: jest.fn(),
  config: {},
};

describe('<Application />', () => {
  it('render without crashing', () => {
    const testId = 'applicationContainer';
    const wrapper = enhancedRender(
      <Application
        data-testid={testId}
        appCheckConfig={defaultProps.appCheckConfig}
        config={defaultProps.config}
      />,
    );
    const {getByTestId, container} = wrapper;
    expect(getByTestId(testId)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
