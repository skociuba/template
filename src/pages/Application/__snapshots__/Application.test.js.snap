// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Application /> render without crashing 1`] = `
<div
  data-testid="applicationContainer"
>
   
  <div
    class="mdc-layout-grid"
  >
    <div
      class="mdc-layout-grid__inner"
    >
      
      <div
        class="mdc-layout-grid__cell mdc-layout-grid__cell--span-8"
      >
        <div
          class="template-1um37w"
        >
          <button
            class="mdc-button"
          >
            <span
              class="mdc-button__label"
            >
              |||
            </span>
          </button>
          <div />
        </div>
      </div>
      <div
        class="mdc-layout-grid__cell"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
